target/linux/pistachio/base-files/etc/init.d/bt-config

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh /etc/rc.common
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+START=85
+STOP=85
+
+config_bluetooth() {
+	. /lib/functions/mac.sh
+	. /lib/functions/bluetooth.sh
+
+	local IS_BT_RUNNING
+	local timeout=10
+	local time=`awk -F "." '{print $1}' /proc/uptime | awk '{print $1}'`
+	local elap_time=0
+
+	while [ -z "$IS_BT_RUNNING" ]
+	do
+		if [ $elap_time -lt $timeout ]
+		then
+			elap_time=$( elapsed_time $time )
+		else
+			logger " waiting for bluetoothd daemon: bt-config timed out "
+			return 1
+		fi 
+		IS_BT_RUNNING=`ps | grep bluetoothd | grep -v grep`
+	done
+	# give some time for the bluetoothd to come up before applying 
```

Comment:
The point of the while loop is to not apply config before the bluetoothd daemon starts, the bluetoothd although is executed before this script but the procd system takes its own time and generally executed a bit later, the sleep here is for giving time to the bluetoothd daemon, to start all the profiles, read the configuration, and find the controller, if we try to config a bit earlier the bluetoothd does not find controller as hciconfig reset in this script brings the interface down. The only way I can think of to find if the bluetoothd has found the controller is to enable the logging of the bluetoothd and search for controller found string, but enabling the log is not good option.